tests/snmp/test_snmp_interfaces.py

```diff
@@ -139,12 +139,13 @@ def verify_snmp_speed(facts, snmp_facts, results):
 def test_snmp_interfaces(localhost, creds_all_duts, duthosts, enum_rand_one_per_hwsku_hostname, enum_asic_index):
     """compare the snmp facts between observed states and target state"""
     duthost = duthosts[enum_rand_one_per_hwsku_hostname]
-    if duthost.is_supervisor_node():
-        pytest.skip("interfaces not present on supervisor node")
     hostip = duthost.host.options['inventory_manager'].get_host(duthost.hostname).vars['ansible_host']
 
     namespace = duthost.get_namespace_from_asic_id(enum_asic_index)
     config_facts  = duthost.config_facts(host=duthost.hostname, source="persistent", namespace=namespace)['ansible_facts']
+    if 'PORT' not in config_facts:
```

Comment:
as discussed, instead of skipping the test. 
we can check if can populate the `snmp_ifnames` only if the snmp_facts is not empty?

Reply:
Not skipping any test, will run all tests for supervisor to ensure that snmp_facts are retrieved for supervisor and linecards.
In supervisor, only mgmt interface will be present in snmp_ifnames.